File: source/aoBlogs2/Views/LatestPostWidget.cs
using System;
using Contensive.Addons.Blog.Models;
using Contensive.Addons.Blog.Models.Db;
using Contensive.BaseClasses;
using Contensive.DesignBlockBase.Controllers;

namespace Contensive.Addons.Blog {

    public class LatestPostWidget : AddonBaseClass {

        public override object Execute(CPBaseClass cp) {
            try {
                string widgetName = "Latest Posts Widget";
                return DesignBlockController.renderWidget<DbLatestPostsWidgetsModel, LatestPostWidgetViewModel>(cp,
                    widgetName: widgetName, 
                    layoutGuid: "{987cb36b-22f8-4896-a54e-aa7dbab98f93}", 
                    layoutName: "Latest Posts Layout", 
                    layoutPathFilename: @"aoBlog\LatestPostLayout.html", 
                    layoutBS5PathFilename: @"aoBlog\LatestPostLayout.html");
            }
            catch (Exception ex) {
                cp.Site.ErrorReport(ex);
                return "";
            }
        }
    }
}

File: source/aoBlogs2/Views/LatestPostWidget.vb
Imports Contensive.BaseClasses

Public Class LatestPostWidget
    Inherits AddonBaseClass

    Public Overrides Function Execute(ByVal cp As CPBaseClass) As Object
        Try
            Dim widgetName As String = "Latest Post Widget"
            Return DesignBlockController.renderWidget(Of DbLatestNewsWidgetsModel, LatestNewsWidgetViewModel)(cp, widgetName:=widgetName, layoutGuid:="{cf55cad2-a7a4-4c74-8ffe-4327e3174372}", layoutName:="Lastest Post Layout", layoutPathFilename:="LatestPostLayout.html", layoutBS5PathFilename:="LatestPostLayout.html")
        Catch ex As Exception
            cp.Site.ErrorReport(ex)
            Return ""
        End Try
    End Function
End Class
